cfme/tests/storage/test_volume.py

```diff
@@ -92,29 +134,62 @@ def test_storage_volume_crud(appliance, provider):
         casecomponent: Cloud
     """
     # create volume
-    volume_collection = appliance.collections.volumes
-    manager_name = '{} Cinder Manager'.format(provider.name)
-    volume = volume_collection.create(name=fauxfactory.gen_alpha(),
-                                      storage_manager=manager_name,
-                                      tenant=provider.data['provisioning']['cloud_tenant'],
-                                      size=STORAGE_SIZE,
-                                      provider=provider)
-    assert volume.exists
+    volume = create_volume(appliance, provider, from_manager, should_assert=True)
 
     # update volume
     old_name = volume.name
     new_name = fauxfactory.gen_alpha()
-    with update(volume):
-        volume.name = new_name
-
-    with update(volume):
-        volume.name = old_name
+    updates = {'volume_name': new_name, 'volume_size': STORAGE_SIZE + 1}
+    if provider.one_of(OpenStackProvider):
+        updates = {'volume_name': new_name}
+
+    if from_manager:
+        storage_manager = appliance.collections.block_managers.filter(
+            {"provider": provider}).all()[0]
+    else:
+        storage_manager = None
+    volume = volume.update(updates, storage_manager)
+    if provider.one_of(EC2Provider):
+        wait_for(lambda: volume.size == '{} GB'.format(updates.get('volume_size')), delay=15,
+                 timeout=900)
+
+    updates = {'volume_name': old_name}
+    volume = volume.update(updates, storage_manager)
 
     # delete volume
-    volume.delete(wait=True)
+    volume.delete(wait=True, storage_manager=storage_manager)
     assert not volume.exists
 
 
+@pytest.mark.uncollectif(lambda appliance, from_manager: appliance.version < '5.10' and
```

Comment:
Please remove check as above.

Reply:
@digitronik What do you think? I think property is named size and attribute is named volume_size?

Reply:
@mmojzis above check l.e. `uncollectif` was applicable for appliance less than `5.10` means `5.9`.